### PR TITLE
LL-17:move old backend code

### PR DESCRIPTION
**Pull request by dampbacon**
Old backend code cleaned up a bit. Some minor refactoring may need to be done for consistency. The db model is still useful only really need to make additions for users and tags. I included the old sqlite db, with some hardcoded routes, so people can mess around with it before we migrate db elsewhere/make sweeping changes.

### CODE REVIEW

**Comment by Exotica0122** [comment]
Why the necessity to nest the project? The root directory should be the main project not in another folder. Now you have `local-lattes-backend/local-lattes-backend`.
The whole point of separating out the repos was for this point.

Just put the files on the root folder

File: local-lattes-backend/cafeRecAPI.sln
Comment (Exotica0122) [inline local-lattes-backend/cafeRecAPI.sln:1]: Should be git ignored. Don't want IDE settings on the repo. If you remove the nesting it should resolve

File: local-lattes-backend/cafeRecAPI/CafeDB/CafeDB.db
Comment (Exotica0122) [inline local-lattes-backend/cafeRecAPI/CafeDB/CafeDB.db:1]: Don't want local DB file please. Please clean up....

File: local-lattes-backend/cafeRecAPI/Controllers/CafeApiController.cs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+{
+    [ApiController]
+    [Route("api/[controller]")]
+    public class CafeApiController : ControllerBase
+    {
+        private readonly ILogger<CafeApiController> _logger;
+        private readonly ICafeService cafeService;
+        private readonly ICafeRepo _repository;
+
+
```
Comment (Exotica0122) [inline local-lattes-backend/cafeRecAPI/Controllers/CafeApiController.cs]: Remove whitespace

```suggestion
```

Comment (Exotica0122) [reply local-lattes-backend/cafeRecAPI/Controllers/CafeApiController.cs]: There's lot's of whitespace in this file. Please clean up

File: local-lattes-backend/cafeRecAPI/Controllers/CafeApiController.cs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+            _logger = logger;
+            cafeService = CafeService;
+            _repository = repository;
+        }
+
+        [HttpGet("GetCafeList")]
+        public async Task<List<Cafe>> GetCafeList(string location)
+        {
+            var cafes = await cafeService.GetCafesAsync(location);
+            return cafes;
```
Comment (Exotica0122) [inline local-lattes-backend/cafeRecAPI/Controllers/CafeApiController.cs]: Could oneline
```suggestion
            return await cafeService.GetCafesAsync(location);
```

Comment (Exotica0122) [reply local-lattes-backend/cafeRecAPI/Controllers/CafeApiController.cs]: Can clean up the business logic in a separate PR. But plz clean up whitespaces

File: local-lattes-backend/cafeRecAPI/Controllers/CafeApiController.cs [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        {
+            var osInfo = new ServerOsInfo
+            {
+                OsDescription = RuntimeInformation.OSDescription,
+                OsArchitecture = RuntimeInformation.OSArchitecture,
+                FrameworkDescription = RuntimeInformation.FrameworkDescription
+            };
+
+            return Ok(osInfo);
+        }
```
Comment (Exotica0122) [inline local-lattes-backend/cafeRecAPI/Controllers/CafeApiController.cs]: umm why do you need this?

**Comment by dampbacon** [comment]
it was an oversight

File: local-lattes-backend/cafeRecAPI/Data/CafeDBContext.cs [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+	{
+        public CafeDBContext(DbContextOptions<CafeDBContext> options) : base(options) { }
+
+        public DbSet<Review> Reviews { get; set; }
+        public DbSet<Cafe> Cafes { get; set; }
+        public DbSet<Tag> Tags { get; set; }
+        public DbSet<SearchLocation> Locations { get; set; }
+        public DbSet<CafeSearchLocation> CafeSearchLocations { get; set; }
+
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
```
Comment (Exotica0122) [inline local-lattes-backend/cafeRecAPI/Data/CafeDBContext.cs]: Weird naming OnCafeCreated` would be better. Don't name variable/functions with "-ing" what does that mean? Does it mean it's created or it's creating? and if it's "creating" when is it finished created.

File: local-lattes-backend/cafeRecAPI/Data/CafeDBContext.cs [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+        {
+            optionsBuilder.UseLazyLoadingProxies();
+        }
+
+    }
+
+
+}
```
Comment (Exotica0122) [inline local-lattes-backend/cafeRecAPI/Data/CafeDBContext.cs]: Try to add whitespace after each file or github won't like it
```suggestion
}

```

File: local-lattes-backend/cafeRecAPI/Data/CafeRepo.cs
Comment (Exotica0122) [inline local-lattes-backend/cafeRecAPI/Data/CafeRepo.cs:1]: What's up with the indentation with this file?

File: local-lattes-backend/cafeRecAPI/Dockerfile
Comment (Exotica0122) [inline local-lattes-backend/cafeRecAPI/Dockerfile:1]: Please don't add dockerfile here. This PR is scoped for cleaning up the old repo. Your increasing the scope and increasing load for code reviewers. Do it in a separate ticket and PR please

File: local-lattes-backend/cafeRecAPI/Utils/makeTags.cs
Comment (Exotica0122) [inline local-lattes-backend/cafeRecAPI/Utils/makeTags.cs:1]: What? just delete this file if it has nothing

**Review by Exotica0122 — COMMENTED** [review]
Some files should be .gitignored.
Also please fix the weird indentations and whitespaces

**Comment by Exotica0122** [comment]
Bro you're still nesting the project. The Controllers should be at the root of the project. We want a flat folder structure
Do you understand what I mean? Right now you have a project inside a another project. Previously it was a project inside a project inside a project (3 levels deep)

File: local-lattes-backend/cafeRecAPI/Dockerfile
Comment (dampbacon) [reply local-lattes-backend/cafeRecAPI/Dockerfile:1]: this is from the old repo, i can remove it if you want.

File: local-lattes-backend/cafeRecAPI/Controllers/CafeApiController.cs (hunk 3 above)
Comment (dampbacon) [reply local-lattes-backend/cafeRecAPI/Controllers/CafeApiController.cs]: was messing around with docker configuration, at some point, i will remove.

File: local-lattes-backend/cafeRecAPI/Data/CafeDBContext.cs (hunk 4 above)
Comment (dampbacon) [reply local-lattes-backend/cafeRecAPI/Data/CafeDBContext.cs]: it's the rules for any model relationship, not just cafe, unsure what else to name it

File: local-lattes-backend/cafeRecAPI/Controllers/CafeApiController.cs (hunk 3 above)
Comment (Exotica0122) [reply local-lattes-backend/cafeRecAPI/Controllers/CafeApiController.cs]: Just remove it now you don't need this for docker

File: local-lattes-backend/cafeRecAPI/Dockerfile
Comment (Exotica0122) [reply local-lattes-backend/cafeRecAPI/Dockerfile:1]: Yea just remove it for this PR

**Comment by dampbacon** [comment]
forgot to remove .sln,
it's a text file that groups together multiple project files.
not really an ide configuration.
visual studio complains if i touch that or vxcproj (compilation files)

**Comment by dampbacon** [comment]
*.csproj

File: Controllers/CafeApiController.cs [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+using cafeRecAPI.Service;
+
+namespace cafeRecAPI.Controllers
+{
+    [ApiController]
+    [Route("api/[controller]")]
+    public class CafeApiController : ControllerBase
+    {
+        private readonly ILogger<CafeApiController> _logger;
+        private readonly ICafeService cafeService;
```
Comment (Exotica0122) [inline Controllers/CafeApiController.cs]: Good standards to prefix private fields with underscore
```suggestion
        private readonly ICafeService _cafeService;
```

File: DTO/CafeDto.cs
Comment (Exotica0122) [inline DTO/CafeDto.cs:1]: Is it just me or is your IDE setup a bit weird. You have 2 space indenting in some files and 4 spaces in some other

File: Data/CafeRepo.cs [hunk 7]
```diff
@@ -0,0 +1,59 @@
+//implement
```
Comment (Exotica0122) [inline Data/CafeRepo.cs:1]: Do to put TODOs. Some IDEs have built in methods to parse through TODOs and find them
```suggestion
// @TODO: description...
```

File: DTO/CafeDto.cs
Comment (dampbacon) [reply DTO/CafeDto.cs:1]: on my ide it looks all the same, any files in particular you noticed?

File: Data/CafeRepo.cs (hunk 7 above)
Comment (dampbacon) [reply Data/CafeRepo.cs:1]: ill go throught tomorrow and edit the relevant comments

**Comment by dampbacon** [comment]
Somthing to note is that the program does not compile in it's current state due to missing db directory (not the db).
Hopefully all you need to do is modify the connection so it creates a db based on the existing models once setup with the remote thing.

Further more onto the spacing thing, not sure if you saw above but it looks the same in my IDE. Only thing i see it kinda weird is that it looks different here on github. I'll modify the files i see here.

**Comment by dampbacon** [comment]
fuck changing the spacing here then commiting reverts it to 8 spaces for some reason

**Review by Exotica0122 — APPROVED** [review]
Gucci for now, merge it when you are ready

**Pull request merged**